sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.controller.getInProgressApplicationDetails.e2e-spec.ts

```diff
@@ -0,0 +1,277 @@
+      .expect(HttpStatus.OK)
+      .expect({
+        id: application.id,
+        applicationStatus: application.applicationStatus,
+        pirStatus: application.pirStatus,
+        studentIncomeVerificationStatus: SuccessWaitingStatus.Success,
+        assessmentInCalculationStep: SuccessWaitingStatus.Success,
+      });
+  });
+
+  it("Should get application in-progress details when the student has parents as supporting users.", async () => {
+    // Arrange
+
+    const application = await saveFakeApplication(db.dataSource, { student });
+
+    // Create supporting users.
+    const parent1 = createFakeSupportingUser(
+      { application },
+      {
+        initialValues: {
+          supportingUserType: SupportingUserType.Parent,
+          supportingData: { totalIncome: 2000 },
+        },
+      },
+    );
+    const parent2 = createFakeSupportingUser(
+      { application },
+      {
+        initialValues: {
+          supportingUserType: SupportingUserType.Parent,
+          supportingData: { totalIncome: 2000 },
+        },
+      },
+    );
+    await db.supportingUser.save([parent1, parent2]);
+
+    // Create CRA income verifications for student, parent1 and parent2.
+    const studentCRAIncomeVerification = createFakeCRAIncomeVerification(
+      {
+        application,
+      },
+      { initialValues: { dateReceived: new Date() } },
+    );
+    const parent1CRAIncomeVerification = createFakeCRAIncomeVerification(
+      {
+        application,
+        supportingUser: parent1,
+      },
+      { initialValues: { dateReceived: new Date() } },
+    );
+    const parent2CRAIncomeVerification = createFakeCRAIncomeVerification(
+      {
+        application,
+        supportingUser: parent2,
+      },
+      { initialValues: { dateReceived: new Date() } },
+    );
+    await db.craIncomeVerification.save([
+      studentCRAIncomeVerification,
+      parent1CRAIncomeVerification,
+      parent2CRAIncomeVerification,
+    ]);
+
+    const endpoint = `/students/application/${application.id}/in-progress`;
+    const token = await getStudentToken(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+    );
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK)
+      .expect({
+        id: application.id,
+        applicationStatus: application.applicationStatus,
+        pirStatus: application.pirStatus,
+        studentIncomeVerificationStatus: SuccessWaitingStatus.Success,
+        parent1IncomeVerificationStatus: SuccessWaitingStatus.Success,
+        parent2IncomeVerificationStatus: SuccessWaitingStatus.Success,
+        parent1Info: SuccessWaitingStatus.Success,
+        parent2Info: SuccessWaitingStatus.Success,
+        assessmentInCalculationStep: SuccessWaitingStatus.Success,
+      });
+  });
+
+  it("Should get application in-progress details when the student has partner as supporting user.", async () => {
```

Comment:
Highly appreciate your effort here. But these E2E tests are not part of the ticket. 

I would advise to keep the efforts of E2E relevant to the user story.

I was expecting an E2E test with and without first outstanding assessment.